<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>wSIR workshop • BioCAsia2024wSIR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="wSIR workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioCAsia2024wSIR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/wSIR_workshop.html">wSIR workshop</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/BioCAsia_2024_wSIR">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>wSIR workshop</h1>
                        <h4 data-toc-skip class="author">Max Woollard<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/SydneyBioX/BioCAsia_2024_wSIR/blob/devel/vignettes/wSIR_workshop.Rmd"><code>vignettes/wSIR_workshop.Rmd</code></a></small>
      <div class="hidden name"><code>wSIR_workshop.Rmd</code></div>

    </div>

    
    
<style>
.question {
  padding: 1em;
  background: lightcyan;
  color: black;
  border-radius: 10px;
}
</style>
<div class="section level2">
<h2 id="wsir-weighted-sliced-inverse-regression-for-supervised-dimension-reduction-of-spatial-transcriptomics-and-single-cell-gene-expression-data">wSIR: Weighted Sliced Inverse Regression for supervised dimension
reduction of spatial transcriptomics and single cell gene expression
data<a class="anchor" aria-label="anchor" href="#wsir-weighted-sliced-inverse-regression-for-supervised-dimension-reduction-of-spatial-transcriptomics-and-single-cell-gene-expression-data"></a>
</h2>
<p><strong>Authors</strong></p>
<p>P = presenting, C = contributing to work</p>
<p>Max Woollard
(P)<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2,3}</annotation></semantics></math>,
Pratibha Panwar
(C)<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2,3}</annotation></semantics></math>,
Linh Nghiem
(C)<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2}</annotation></semantics></math>,
Shila Ghazanfar
(C)<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn></mrow></msup><annotation encoding="application/x-tex">^{1,2,3}</annotation></semantics></math>.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>
Sydney Precision Data Science Centre, University of Sydney,
Australia<br><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>
School of Mathematics and Statistics, University of Sydney, Australia
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>3</mn></msup><annotation encoding="application/x-tex">^3</annotation></semantics></math>
Charles Perkins Centre, University of Sydney, Australia<br></p>
<p><br> Contact: mwoo5086@uni.sydney.edu.au</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>Single-cell and spatial gene expression data are very
high-dimensional, which makes dimension reduction a crucial step in the
analysis pipeline. However, many popular examples of dimension reduction
methods, such as Principal Component Analysis (PCA), ignore the purpose
of a dimension reduction procedure, which is to assist in a specific
downstream analysis. wSIR is a supervised dimension reduction method,
which makes use of cells’ spatial locations to compute a
spatially-informed low-dimensional embedding.</p>
</div>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>In this workshop, we will introduce the wSIR method for supervised
dimension reduction of spatial transcriptomics and single-cell gene
expression data. We will explain the method, and go through some example
analyses to demonstrate how the method is applied. We will also perform
some simple analyses to demonstrate its utility.</p>
<div class="section level4">
<h4 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h4>
<p>It is expected that students will have:</p>
<ul>
<li>basic knowledge of R syntax, and</li>
<li>familiarity with single-cell and spatial transcriptomics data.</li>
</ul>
</div>
<div class="section level4">
<h4 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h4>
<p>The expected timing of the workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>wSIR method overview</td>
<td>20m</td>
</tr>
<tr class="even">
<td>wSIR application: interpretability</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>wSIR application: improved modelling accuracy</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<ul>
<li>Understand the wSIR method.</li>
<li>Use wSIR to project new single-cell gene expression data into a
spatially-informed low-dimensional space.</li>
<li>Examine the wSIR loadings to derive biological insight</li>
<li>Use wSIR to improve accuracy in downstream analysis</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="workshop">Workshop<a class="anchor" aria-label="anchor" href="#workshop"></a>
</h3>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<div class="section level4">
<h4 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#install.packages(c("ggplot2", "vctrs", "magrittr", "devtools"))</span></span>
<span><span class="co">#devtools::install_github("SydneyBioX/wSIR")</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SydneyBioX/BioCAsia_2024_wSIR">BioCAsia2024wSIR</a></span><span class="op">)</span>  <span class="co"># use the same name, no underscores, as in DESCRIPTION</span></span>
<span><span class="co">#&gt; Loading required package: wSIR</span></span>
<span><span class="co">#&gt; Loading required package: magrittr</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span><span class="co">#&gt; Loading required package: vctrs</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vctrs.r-lib.org/" class="external-link">vctrs</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sydneybiox.github.io/wSIR" class="external-link">wSIR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span> <span class="co"># for arrange</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:vctrs':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     data_frame</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="acquire-data">Acquire data<a class="anchor" aria-label="anchor" href="#acquire-data"></a>
</h4>
<p>We will use spatial transcriptomics data for mouse embryos from <a href="https://www.nature.com/articles/s41587-021-01006-2" class="external-link uri">https://www.nature.com/articles/s41587-021-01006-2</a> . We
will examine how we can apply the wSIR functions to study this data.
This dataset will illustrate how you can apply the package functions to
your own data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#data(embryos_data_red)  # you don't have a data folder</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"embryos_data_red.RData"</span>, package<span class="op">=</span><span class="st">"BioCAsia2024wSIR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## files this downloads:</span></span>
<span><span class="co"># exprs1</span></span>
<span><span class="co"># coords1</span></span>
<span><span class="co"># exprs2</span></span>
<span><span class="co"># coords2</span></span>
<span><span class="co"># exprs3</span></span>
<span><span class="co"># coords3</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure coordinate column names are correct</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">coords1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">coords2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">coords3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">exprs1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exprs1</span><span class="op">)</span></span>
<span><span class="va">exprs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exprs2</span><span class="op">)</span></span>
<span><span class="va">exprs3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exprs3</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exprs1</span><span class="op">)</span></span>
<span><span class="va">n2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exprs2</span><span class="op">)</span></span>
<span><span class="va">n3</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">exprs3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">n1</span>, <span class="va">n2</span>, <span class="va">n3</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  9792  7372 11632</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="wsir-method">wSIR method<a class="anchor" aria-label="anchor" href="#wsir-method"></a>
</h3>
<p>wSIR is an extension of the supervised dimension reduction technique
of Sliced Inverse Regression (SIR).</p>
<p>SIR is an existing supervised dimension reduction method which works
by grouping together the observations with similar values for the
response. For spatial transcriptomics data, this means grouping all the
cells into a certain number of tiles based on their spatial position.
For example, if we use 4 tiles, then the cells in the top right quadrant
of the image/tissue go to one group, those in the top left to another,
and so on. Each of those groups is then summarised by averaging the
expression level of all cells in each group for each of the genes. From
there, eigendecomposition is performed on the resulting matrix of
tile-gene means, then returning the SIR directions and SIR scores.</p>
<p>The motivation behind wSIR is that SIR only uses each cell’s spatial
position when we are separating the cells into the given number of
groups/tiles. Once those groups are created, we lose the fact that some
groups may be more spatially related (if they come from adjacent tiles)
than other groups (if they come from opposite sides of the tissue). wSIR
uses a weight matrix to incorporate the spatial correlation between all
pairs of cells in the SIR algorithm. This matrix has dimension H*H,
where H is the number of tiles, and the (i,j)th entry represents the
distance between tiles i and j. This matrix is incorporated into the
eigendecomposition step. The wSIR output has the same structure as the
SIR output.</p>
</div>
<div class="section level3">
<h3 id="method-demonstration">Method demonstration<a class="anchor" aria-label="anchor" href="#method-demonstration"></a>
</h3>
<p>We will begin the workshop by going through some steps in the wSIR
algorithm. You do not need to understand the steps in the method to be
able to use the package - we only include this to give you an
understanding of how it works</p>
<div class="section level4">
<h4 id="tile-creation">Tile creation<a class="anchor" aria-label="anchor" href="#tile-creation"></a>
</h4>
<p>The key conceptual difference between SIR and wSIR is in how we
estimate E(X|Y).</p>
<p>In both cases, this begins by grouping the cells into tiles according
to their spatial positions. We split the data into rectangular tiles,
with an equal number of “cuts” in both spatial axes.</p>
<p>We will work here on the spatial coordinates of embryo 3, as this is
the sample with the most cells.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">coords3</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"X spatial coordinate"</span>, y <span class="op">=</span> <span class="st">"Y spatial coordinate"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Embryo 3 spatial coordinates"</span><span class="op">)</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>We will then split the data into 9 tiles.</p>
<p>This means assigning each cell to one tile of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">(1,1), (1,2), (1,3), (2,1), (2,2), (2,3), (3,1), (3,2), (3,3)</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(1,1)</annotation></semantics></math>
is the tile in the top left,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">(1,3)</annotation></semantics></math>
is in the top right, and so on. We do this by assigning all cells in the
upper third of the sample (according to spatial position) to one of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">(1,1), (1,2), (1,3)</annotation></semantics></math>,
then the middle third to one of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">(2,1), (2,2), (2,3)</annotation></semantics></math>,
and the lower third to one of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>,</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">(3,1), (3,2), (3,3)</annotation></semantics></math>.
This is done so that each of those thirds is as equally populated as
possible. Next, a similar procedure is applied to the left third, the
middle third, and the right third.</p>
<div class="question">
<p><strong>Questions</strong></p>
<ol style="list-style-type: decimal">
<li>Is this procedure ensuring an equal number of cells in all
tiles?</li>
<li>Try modifying the chunk below to allocate the cells to tiles in
different ways.</li>
</ol>
</div>
<p>Note the code to do this in the wSIR package has been optimised - the
code here is only to demonstrate the concept.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define the number of cells in each third of the sample</span></span>
<span></span>
<span><span class="va">third1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="va">n3</span><span class="op">/</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">third2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="va">n3</span><span class="op">/</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">third3</span> <span class="op">&lt;-</span> <span class="va">n3</span> <span class="op">-</span> <span class="va">third1</span> <span class="op">-</span> <span class="va">third2</span></span>
<span></span>
<span><span class="co"># create new dataframe with row IDs, which we sort by later </span></span>
<span><span class="va">with_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">coords3</span><span class="op">$</span><span class="va">x</span>,</span>
<span>                      y <span class="op">=</span> <span class="va">coords3</span><span class="op">$</span><span class="va">y</span>,</span>
<span>                      id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">n3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># sort by x spatial coordinate to find the tile coordinates in the x spatial axis</span></span>
<span><span class="va">x_sorted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">with_id</span>, <span class="va">x</span><span class="op">)</span></span>
<span><span class="va">x_sorted</span><span class="op">$</span><span class="va">x_tile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">third1</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">third2</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">third3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># sort by y spatial coordinate to find the tile coordinates in the y spatial axis</span></span>
<span><span class="va">y_sorted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">x_sorted</span>, <span class="va">y</span><span class="op">)</span></span>
<span><span class="va">y_sorted</span><span class="op">$</span><span class="va">y_tile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">third1</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">third2</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">third3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># sort data back into original order so the tile_allocations vector is in the same order as the expression and coordinates data</span></span>
<span><span class="va">id_sorted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">y_sorted</span>, <span class="va">id</span><span class="op">)</span></span>
<span><span class="va">tile_allocations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">id_sorted</span><span class="op">$</span><span class="va">x_tile</span>, <span class="st">", "</span>, <span class="va">id_sorted</span><span class="op">$</span><span class="va">y_tile</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot cells in spatial positions coloured by tile allocation</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">coords3</span><span class="op">$</span><span class="va">x</span>, </span>
<span>                y <span class="op">=</span> <span class="va">coords3</span><span class="op">$</span><span class="va">y</span>, </span>
<span>                tile <span class="op">=</span> <span class="va">tile_allocations</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span>, colour <span class="op">=</span> <span class="va">tile</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"X spatial axis"</span>, y <span class="op">=</span> <span class="st">"Y spatial axis"</span>, colour <span class="op">=</span> <span class="st">"Tile alocation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Cells from embryo 3 coloured by tile allocation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="weight-matrix-creation">Weight matrix creation<a class="anchor" aria-label="anchor" href="#weight-matrix-creation"></a>
</h4>
<p>Once the cells are allocated to tiles, the weight matrix is then
created. For this workshop, we will use the
<code>CreateWeightMatrix</code> function from the wSIR package. Note
this is a hidden function as users do not need to use it, but we will
access it anyway for this workshop.</p>
<p>You can adjust the alpha parameter, which represents the strength of
spatial weighting, and then visualise the resulting weight matrix.</p>
<div class="question">
<p><strong>Questions</strong></p>
<ol style="list-style-type: decimal">
<li>Observe what happens when you have 0 spatial weighting, or very
strong spatial weighting.</li>
<li>If you had never heard of wSIR and were asked to quantify the
spatial proximity of the tiles above in a matrix with values between 0
and 1, how would it look? Is there a value of alpha that would produce
such a matrix?</li>
</ol>
</div>
<p>If you want to adjust the tile allocations, it is best to do that in
the previous chunk. Then you can simply run this next chunk to view the
resulting weight matrix.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Modify tile allocations to suit the required input for CreateWeightMatrix function.</span></span>
<span><span class="co"># As with other components in this section, you won't need to know this to use the wSIR package - it is just included in case you are very curious about how the functions work. </span></span>
<span></span>
<span><span class="co"># requires dataframe input with a column named "coordinate". Tile allocations must have 3 "axes": one for each spatial axis, and a third for the sample each cell belongs to. In this case, all cells are from the same sample, which is why I specify 1 in the next line. This would change if we used multiple samples at once, in which case we need to set the weight value between tiles of differing samples to 0.</span></span>
<span></span>
<span><span class="va">tile_allocations_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>coordinate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">tile_allocations</span>, <span class="st">", 1"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">weight_matrix</span> <span class="op">&lt;-</span> <span class="fu">wSIR</span><span class="fu">:::</span><span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/createWeightMatrix.html" class="external-link">createWeightMatrix</a></span><span class="op">(</span>coords <span class="op">=</span> <span class="va">coords3</span>, </span>
<span>                                           labels <span class="op">=</span> <span class="va">tile_allocations_df</span>, </span>
<span>                                           alpha <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="co">## Adjust this parameter!</span></span>
<span></span>
<span><span class="va">n_tiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">weight_matrix</span><span class="op">)</span></span>
<span><span class="co"># Turn weight matrix into a format that can be visualised in a ggplot heatmap</span></span>
<span></span>
<span><span class="va">w_mat_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">weight_matrix</span><span class="op">)</span>,</span>
<span>                       x <span class="op">=</span> <span class="fu"><a href="https://vctrs.r-lib.org/reference/vec-rep.html" class="external-link">vec_rep_each</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">n_tiles</span><span class="op">)</span>, <span class="va">n_tiles</span><span class="op">)</span>,</span>
<span>                       y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">n_tiles</span><span class="op">)</span>, <span class="va">n_tiles</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">w_mat_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Tile"</span>, y <span class="op">=</span> <span class="st">"Tile"</span>, fill <span class="op">=</span> <span class="st">"Weight matrix value"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Example weight matrix"</span><span class="op">)</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>The next steps in the wSIR method include more complicated linear
algebra, which we will not cover in this workshop.</p>
<p>This is the end of the method demonstration section. Up to now, we
hope you have an understanding of how the tiles are allocated, and how
the weight matrix will look with varying allocations and spatial
weighting strengths.</p>
</div>
</div>
<div class="section level3">
<h3 id="wsir-parameter-exploration">wSIR parameter exploration<a class="anchor" aria-label="anchor" href="#wsir-parameter-exploration"></a>
</h3>
<p>Here, we demonstrate the wSIR function
<code>exploreWSIRParams</code>, which shows the impact of changing the
parameter values on the distance correlation between the wSIR
low-dimensional embedding and the true spatial coordinates. A higher
distance correlation corresponds to a better performance.</p>
<p>Distance correlation is a measure of the correlation between two
distance matrices. Ideally, if two cells are spatially close, we would
want them to be close in the low-dimensional embedding, and similar for
two distant cells - distance correlation is a measure of this. If you’d
like more details on this metric, there are many good online
resources.</p>
<p>The following chunk has some parameters we you can explore. You can
vary the alpha and slice values that are tuned over, to see the
performance with different combinations. This can take a while to run,
so we suggest you randomly subset some of the data with the variable
<code>subsetted</code>. In practice, this doesn’t change the results of
your study, we only do it as computing the distance correlation can be
quite slow over many cells.</p>
<p>We recommend you don’t adjust <code>nrep</code> or
<code>varThreshold</code>, as this can make it take a very long time to
run. Be aware that the following chunk can take some time to run, so be
careful with what you change!</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subsetted</span> <span class="op">=</span> <span class="fl">0.2</span> <span class="co"># Change this to specify the proportion of the data you want to use for this exploration</span></span>
<span><span class="va">rsample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>, size <span class="op">=</span> <span class="va">n3</span>, replace <span class="op">=</span> <span class="cn">TRUE</span>, prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">subsetted</span>, <span class="fl">1</span><span class="op">-</span><span class="va">subsetted</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">EWP_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/exploreWSIRParams.html" class="external-link">exploreWSIRParams</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">exprs3</span><span class="op">[</span><span class="va">rsample</span>,<span class="op">]</span>,</span>
<span>                                coords <span class="op">=</span> <span class="va">coords3</span><span class="op">[</span><span class="va">rsample</span>,<span class="op">]</span>,</span>
<span>                                nrep <span class="op">=</span> <span class="fl">3</span>, <span class="co"># This function computes a random train/test split of the data nrep times</span></span>
<span>                                varThreshold <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>                                optim_alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span>,<span class="fl">10</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                                optim_slices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">10</span>,<span class="fl">15</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; set up nrep random splits of the data into training and test sets</span></span>
<span><span class="co">#&gt; completed runs of wSIR and metric calculation</span></span>
<span><span class="co">#&gt; Optimal (alpha, slices) pair: (20, 5)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">EWP_object</span><span class="op">$</span><span class="va">results_dataframe</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">alpha</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">slices</span>, fill <span class="op">=</span> <span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_fill_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Alpha"</span>, y <span class="op">=</span> <span class="st">"Slices"</span>, fill <span class="op">=</span> <span class="st">"Distance Correlation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>The key takeaway you will (hopefully) see above is that wSIR has
relatively invariant distance correlation for moderate values of slices
and alpha. This is in contrast to the column representing
<code>alpha = 0</code>, which means there is no spatial weighting at all
(recall the corresponding weight matrix). Why do you think performance
deteriorates with an increasing number of slices when
<code>alpha = 0</code>?</p>
</div>
<div class="section level3">
<h3 id="wsir-algorithm">wSIR algorithm<a class="anchor" aria-label="anchor" href="#wsir-algorithm"></a>
</h3>
<p>Using the wSIR algorithm itself is quite straightforward. For users,
there is a single function to compute the low-dimensional embedding and
the loadings matrix, which is named wSIR. This function only needs the
gene expression matrix and the spatial coordinates dataframe as input,
though you can specify parameters to the non-default values if desired.
The wSIR function returns a list of results with 5 (named) slots, whose
details can be found at <code><a href="https://sydneybiox.github.io/wSIR/reference/wSIR.html" class="external-link">?wSIR::wSIR</a></code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># wSIR with default parameter values</span></span>
<span><span class="va">wsir_object</span> <span class="op">&lt;-</span> <span class="fu">wSIR</span><span class="fu">::</span><span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/wSIR.html" class="external-link">wSIR</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">exprs3</span>, </span>
<span>                          coords <span class="op">=</span> <span class="va">coords3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># wSIR selecting to optimise over a range of parameters using exploreWSIRParams</span></span>
<span><span class="co">## Not recommended in practice since wSIR performance is mostly consistent when you select reasonable values for slices and alpha</span></span>
<span><span class="va">wsir_object</span> <span class="op">&lt;-</span> <span class="fu">wSIR</span><span class="fu">::</span><span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/wSIR.html" class="external-link">wSIR</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">exprs3</span>, </span>
<span>                          coords <span class="op">=</span> <span class="va">coords3</span>,</span>
<span>                          optim_params <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                          nrep <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                          optim_alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span>,<span class="fl">10</span><span class="op">)</span>,</span>
<span>                          optim_slices <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">8</span>,<span class="fl">12</span>,<span class="fl">15</span><span class="op">)</span>,</span>
<span>                          varThreshold <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="co">#&gt; set up nrep random splits of the data into training and test sets</span></span>
<span><span class="co">#&gt; completed runs of wSIR and metric calculation</span></span>
<span><span class="co">#&gt; Optimal (alpha, slices) pair: (10, 8)</span></span>
<span></span>
<span><span class="co"># wSIR specifying some parameter values</span></span>
<span><span class="va">wsir_obj</span> <span class="op">&lt;-</span> <span class="fu">wSIR</span><span class="fu">::</span><span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/wSIR.html" class="external-link">wSIR</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">exprs3</span>, </span>
<span>                       coords <span class="op">=</span> <span class="va">coords3</span>,</span>
<span>                       alpha <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                       slices <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                       varThreshold <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="wsir-application-interpretability">wSIR application: interpretability<a class="anchor" aria-label="anchor" href="#wsir-application-interpretability"></a>
</h3>
<p>The wSIR package includes some functions to give an insight into what
biological information the method is using. These functions are firstly
for interpretability, so you can understand which genes are the most
important for the low-dimensional space. Secondly, these functions could
be used to give more biological understanding, as you can find how genes
impact and are impacted by the spatially-aware low-dimensional
embedding.</p>
<div class="section level4">
<h4 id="wsir-top-genes">wSIR Top Genes<a class="anchor" aria-label="anchor" href="#wsir-top-genes"></a>
</h4>
<p>The <code>findTopGenes</code> function finds and plots the genes with
highest loading in the specified wSIR directions (default is direction
1). If a gene has high loading (in terms of magnitude), it is more
important to the wSIR direction. Since the wSIR directions are designed
to retain information about each cell’s spatial position, the genes with
high loading should be spatially-related genes.</p>
<p>In the plot below, we can see which genes have the highest loading in
wSIR direction 1. This is useful as it gives us an intuition about how
wSIR creates the low-dimensional embedding. We can see that some of the
genes are known spatial genes (e.g. Cdx2, Hox-), which is what we would
expect to see.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_genes_obj</span> <span class="op">=</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/findTopGenes.html" class="external-link">findTopGenes</a></span><span class="op">(</span>WSIR <span class="op">=</span> <span class="va">wsir_obj</span>, highest <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="co"># create top genes object</span></span>
<span><span class="va">top_genes_plot</span> <span class="op">=</span> <span class="va">top_genes_obj</span><span class="op">$</span><span class="va">plot</span> <span class="co"># select plot</span></span>
<span><span class="va">top_genes_plot</span> <span class="co"># print plot</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">top_genes_obj</span> <span class="op">=</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/findTopGenes.html" class="external-link">findTopGenes</a></span><span class="op">(</span>WSIR <span class="op">=</span> <span class="va">wsir_obj</span>, highest <span class="op">=</span> <span class="fl">8</span>, dirs <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">top_genes_plot</span> <span class="op">=</span> <span class="va">top_genes_obj</span><span class="op">$</span><span class="va">plot</span></span>
<span><span class="va">top_genes_plot</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="visualising-wsir-scores">Visualising wSIR Scores<a class="anchor" aria-label="anchor" href="#visualising-wsir-scores"></a>
</h4>
<p>The <code>visualiseWSIRDirections</code> function plots each cell at
its spatial position, coloured by its value for each of the specified
wSIR columns. This gives us an understanding of what each column of the
low-dimensional embedding represents.</p>
<p>Below, we visualise the cells at their spatial positions, coloured by
each of the 5 wSIR directions The top left plot illustrates how, for
this example, wSIR direction 1 captures information about the “y”
spatial axis, since cells with higher “y” coordinate have low wSIR1
value, while cells with lower “y” coordinate have higher wSIR1 value.
wSIR2 is shown in the next plot over (the one titled “2”), and we can
see that wSIR column two appears to capture information about the “x”
spatial coordinate. The remaining three wSIR columns all contain
information about cell types, which we can tell by the regions of high
and low wSIR column values spread across the tissue.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vis_obj</span> <span class="op">=</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/visualiseWSIRDirections.html" class="external-link">visualiseWSIRDirections</a></span><span class="op">(</span>coords <span class="op">=</span> <span class="va">coords3</span>, WSIR <span class="op">=</span> <span class="va">wsir_obj</span>, dirs <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> <span class="co"># create visualisations</span></span>
<span><span class="va">vis_obj</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="umap-on-low-dimensional-embedding">UMAP on low-dimensional embedding<a class="anchor" aria-label="anchor" href="#umap-on-low-dimensional-embedding"></a>
</h4>
<p>The two functions <code>generateUmapFromWSIR</code> and
<code>plotUmapFromWSIR</code> create and display UMAP dimension
reduction calculated on the wSIR low-dimensional embedding. We can
colour the UMAP plot (where each point represents a cell) by its value
for various genes of interest. This visualises the structure of the wSIR
dimension reduction space, which is useful to gain more intuition about
what the space represents. Specifically, we can see if the wSIR space
contains neighbourhoods of high expression for specific genes, thus
better understanding how this space is made.</p>
<p>To specify which genes we would like to include, we can use the
output from the <code>findTopGenes</code> function from above, which
finds spatially-related genes by ranking those with the highest loading
in relevant wSIR directions. This output is then the value for the
<code>highest_genes</code> parameter. Otherwise, we could also specify
our own genes of interest if there are some specific genes we would like
to visualise. For example, if we wanted to visualise the expression
distribution for Cdx2 and Hoxb4, we could use
<code>genes = c("Cdx2", "Hoxb4")</code> as an argument in
<code>plotUmapFromWSIR</code> (and leave <code>highest_genes</code>
blank).</p>
<p>Below, we use the UMAP function to visualise the wSIR space computed
on the gene expression data from sample 1. We colour each cell by their
values for the 6 genes with highest value in wSIR direction 1 (as found
by the <code>findTopGenes</code> function previously). We can see that
for some of these genes, there are specific regions of high expression
in the UMAP plots, suggesting that the wSIR space separates cells based
on their expression for those genes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">umap_coords</span> <span class="op">=</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/generateUmapFromWSIR.html" class="external-link">generateUmapFromWSIR</a></span><span class="op">(</span>WSIR <span class="op">=</span> <span class="va">wsir_obj</span><span class="op">)</span></span>
<span><span class="va">umap_plots</span> <span class="op">=</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/plotUmapFromWSIR.html" class="external-link">plotUmapFromWSIR</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">exprs3</span>,</span>
<span>                              umap_coords <span class="op">=</span> <span class="va">umap_coords</span>,</span>
<span>                              highest_genes <span class="op">=</span> <span class="va">top_genes_obj</span>,</span>
<span>                              n_genes <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span><span class="va">umap_plots</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="wsir-application-improving-performance">wSIR application: improving performance<a class="anchor" aria-label="anchor" href="#wsir-application-improving-performance"></a>
</h3>
<p>Aside from improving interpretability and aiding biological insight,
wSIR can improve modelling performance. For users, you may want to
project single-cell gene expression data into a spatially-aware
low-dimensional embedding, to which you can apply any machine learning
methods (e.g cell-type classification, location prediction, etc).</p>
<p>Here, we first demonstrate how to project new single-cell data into a
spatially aware low-dimensional space, using out wSIR object from a
previous section.</p>
<p>We will use the gene expression matrix from mouse embryo 1 here.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exprs1</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9792  351</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exprs3</span><span class="op">)</span> <span class="co"># Note that the genes in your spatial and single-cell samples must be the same, but there can be different numbers of cells.</span></span>
<span><span class="co">#&gt; [1] 11632   351</span></span>
<span></span>
<span><span class="va">exprs1_projected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/projectWSIR.html" class="external-link">projectWSIR</a></span><span class="op">(</span>wsir <span class="op">=</span> <span class="va">wsir_obj</span>, newdata <span class="op">=</span> <span class="va">exprs1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exprs1_projected</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9792   10</span></span></code></pre></div>
<p>From just that line, you can now apply downstream analysis to this
low-dimensional embedding of embryo 1’s gene expression data.</p>
<p>Here, we will show how this method can be applied to Tangram, a
popular deep-learning-based spatial alignment method. Tangram is
available here: <a href="https://www.nature.com/articles/s41592-021-01264-7" class="external-link uri">https://www.nature.com/articles/s41592-021-01264-7</a> .</p>
<p>For this simulation, we will use embryo 3 as our spatial sample, and
embryo 1 as our single-cell sample. The method workflow is detailed
below. The concept will be to use the wSIR low-dimensional embedding to
predict the spatial coordinates of the single-cell data.</p>
<div class="float">
<img src="inst/images/tangram_workflow.png" alt="Tangram downstream analysis example workflow"><div class="figcaption">Tangram downstream analysis example
workflow</div>
</div>
<p>Note that for this workshop, we will not actually compute the Tangram
predicted coordinates, as it takes too long to run. We will load in the
predicted coordinates which we have already computed, and we will create
a results plot from that.</p>
<p>Below loads in 7 matrices, all of dimension n1 by 2, containing the
predicted coordinates using as inputs: PCA, PLS, SIR, wSIR, LDA, counts
and logcounts. The file names are of the form <code>pred_pca_em1</code>,
in that case for the predicted coordinates of embryo 1 using the PCA
low-dimensional embedding as the Tangram input. We also include the
predicted coordinates using just counts or LogCounts as the inputs
(without any dimension reduction applied) as those are the default
inputs for Tangram.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#data(em1_tangram_preds_red) # This loads a list (not vector) of predicted coordinates into your environment, named pred_em1_tangram_red</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"em1_tangram_preds_red.RData"</span>, package<span class="op">=</span><span class="st">"BioCAsia2024wSIR"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>To evaluate, we will compute the distance correlation between the
predicted and the actual coordinates, for the predicted coordinates from
all dimension reduction methods. This is not part of the wSIR package,
but should demonstrate the effectiveness of using wSIR as a dimension
reduction tool to improve downstream analysis.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">wsir_obj</span><span class="op">$</span><span class="va">scores</span><span class="op">)</span> <span class="co"># This is the features matrix for the training set</span></span>
<span><span class="co">#&gt; [1] 11632    10</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">coords3</span><span class="op">)</span> <span class="co"># This is the spatial coordinates for the training set</span></span>
<span><span class="co">#&gt; [1] 11632     2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exprs1_projected</span><span class="op">)</span> <span class="co"># This is the features matrix for the testing set, with which we will use Tangram to predict its spatial coordinates</span></span>
<span><span class="co">#&gt; [1] 9792   10</span></span>
<span></span>
<span><span class="co"># save predicted coordinates in a list for easy results generation</span></span>
<span></span>
<span></span>
<span><span class="va">em1_dcs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">7</span><span class="op">)</span> <span class="co"># initialise empty vector to fill with the distance correlations</span></span>
<span><span class="va">i</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">pred</span> <span class="kw">in</span> <span class="va">pred_em1_tangram_red</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="va">i</span> <span class="op">=</span> <span class="va">i</span> <span class="op">+</span> <span class="fl">1</span></span>
<span>  <span class="va">em1_dcs</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Rfast</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rfast/man/dcor.html" class="external-link">dcor</a></span><span class="op">(</span><span class="va">pred</span>, <span class="va">coords1</span><span class="op">)</span><span class="op">$</span><span class="va">dcor</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; [1] 2</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">#&gt; [1] 4</span></span>
<span><span class="co">#&gt; [1] 5</span></span>
<span><span class="co">#&gt; [1] 6</span></span>
<span></span>
<span><span class="va">tangram_distcor_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>distcor <span class="op">=</span> <span class="va">em1_dcs</span>,</span>
<span>                                 tangram_input <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDA"</span>, </span>
<span>                                                   <span class="st">"wSIR"</span>, </span>
<span>                                                   <span class="st">"SIR"</span>,</span>
<span>                                                   <span class="st">"PCA"</span>,</span>
<span>                                                   <span class="st">"PLS"</span>,</span>
<span>                                                   <span class="st">"Counts"</span>,</span>
<span>                                                   <span class="st">"LogCounts"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tangram_distcor_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tangram_input</span>, y <span class="op">=</span> <span class="va">distcor</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Tangram Input"</span>, y <span class="op">=</span> <span class="st">"Distance Corrrelation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Distance correlation between predicted and actual coordinates"</span><span class="op">)</span></span></code></pre></div>
<p><img src="wSIR_workshop_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="section level4">
<h4 id="wsir-application-multiple-samples">wSIR application: multiple samples<a class="anchor" aria-label="anchor" href="#wsir-application-multiple-samples"></a>
</h4>
<p>In the previous section, we used the gene expression data and the
spatial coordinates of one spatial sample to project the gene expression
data of a different sample into low-dimensional space. However, we may
often have access to multiple spatial samples. wSIR is able to
incorporate multiple such samples, by setting the weight matrix value
for two tiles from different spatial samples to be 0. The code to use,
for example, the gene expression and spatial coordinates for embryos 2
and 3 to project the gene expression data of embryo 1 into
low-dimensional space is below.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wsir_obj_2_3</span> <span class="op">&lt;-</span> <span class="fu">wSIR</span><span class="fu">::</span><span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/wSIR.html" class="external-link">wSIR</a></span><span class="op">(</span>X <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">exprs2</span>, <span class="va">exprs3</span><span class="op">)</span>,</span>
<span>                           coords <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">coords2</span>, <span class="va">coords3</span><span class="op">)</span>,</span>
<span>                           samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">n2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">n3</span><span class="op">)</span><span class="op">)</span>, <span class="co"># specify samples</span></span>
<span>                           alpha <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                           slices <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                           varThreshold <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span><span class="va">proj1_from23</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://sydneybiox.github.io/wSIR/reference/projectWSIR.html" class="external-link">projectWSIR</a></span><span class="op">(</span>wsir <span class="op">=</span> <span class="va">wsir_obj_2_3</span>, newdata <span class="op">=</span> <span class="va">exprs1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">proj1_from23</span><span class="op">)</span> <span class="co"># gene expression data from sample 1 projected into low-dimensional space</span></span>
<span><span class="co">#&gt; [1] 9792   11</span></span></code></pre></div>
<p>We will not explore it here, but our results (manuscript on the way!)
show that wSIR performance improves when you incorporate multiple
spatial samples. Therefore, if you are in a situation where you do have
multiple spatial samples, it is best to use wSIR and account for the
multiple samples as above.</p>
<p>Finish</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Etc/UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] dplyr_1.1.4            BioCAsia2024wSIR_2.0.1 vctrs_0.6.5           </span></span>
<span><span class="co">#&gt; [4] ggplot2_3.5.1          magrittr_2.0.3         wSIR_0.99.3           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] rlang_1.1.4                 matrixStats_1.4.1          </span></span>
<span><span class="co">#&gt;  [3] compiler_4.4.1              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [5] systemfonts_1.1.0           RcppZiggurat_0.1.6         </span></span>
<span><span class="co">#&gt;  [7] stringr_1.5.1               pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [9] SpatialExperiment_1.16.0    crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [11] fastmap_1.2.0               backports_1.5.0            </span></span>
<span><span class="co">#&gt; [13] magick_2.8.5                XVector_0.46.0             </span></span>
<span><span class="co">#&gt; [15] labeling_0.4.3              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt; [17] rmarkdown_2.28              UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt; [19] ragg_1.3.3                  purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [21] xfun_0.48                   Rfast_2.1.0                </span></span>
<span><span class="co">#&gt; [23] zlibbioc_1.52.0             cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [25] GenomeInfoDb_1.42.0         jsonlite_1.8.9             </span></span>
<span><span class="co">#&gt; [27] highr_0.11                  DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt; [29] BiocParallel_1.40.0         Deriv_4.1.6                </span></span>
<span><span class="co">#&gt; [31] broom_1.0.7                 parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [33] R6_2.5.1                    bslib_0.8.0                </span></span>
<span><span class="co">#&gt; [35] stringi_1.8.4               reticulate_1.39.0          </span></span>
<span><span class="co">#&gt; [37] boot_1.3-31                 GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt; [39] jquerylib_0.1.4             Rcpp_1.0.13                </span></span>
<span><span class="co">#&gt; [41] SummarizedExperiment_1.36.0 knitr_1.48                 </span></span>
<span><span class="co">#&gt; [43] modelr_0.1.11               IRanges_2.40.0             </span></span>
<span><span class="co">#&gt; [45] Matrix_1.7-1                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt; [47] abind_1.4-8                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [49] codetools_0.2-20            lattice_0.22-6             </span></span>
<span><span class="co">#&gt; [51] tibble_3.2.1                Biobase_2.66.0             </span></span>
<span><span class="co">#&gt; [53] withr_3.0.2                 askpass_1.2.1              </span></span>
<span><span class="co">#&gt; [55] evaluate_1.0.1              desc_1.4.3                 </span></span>
<span><span class="co">#&gt; [57] RcppParallel_5.1.9          pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [59] MatrixGenerics_1.18.0       stats4_4.4.1               </span></span>
<span><span class="co">#&gt; [61] generics_0.1.3              S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt; [63] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">#&gt; [65] glue_1.8.0                  tools_4.4.1                </span></span>
<span><span class="co">#&gt; [67] doBy_4.6.24                 distances_0.1.11           </span></span>
<span><span class="co">#&gt; [69] RSpectra_0.16-2             fs_1.6.4                   </span></span>
<span><span class="co">#&gt; [71] cowplot_1.1.3               grid_4.4.1                 </span></span>
<span><span class="co">#&gt; [73] tidyr_1.3.1                 umap_0.2.10.0              </span></span>
<span><span class="co">#&gt; [75] colorspace_2.1-1            SingleCellExperiment_1.28.0</span></span>
<span><span class="co">#&gt; [77] GenomeInfoDbData_1.2.13     cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [79] textshaping_0.4.0           fansi_1.0.6                </span></span>
<span><span class="co">#&gt; [81] viridisLite_0.4.2           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt; [83] gtable_0.3.6                sass_0.4.9                 </span></span>
<span><span class="co">#&gt; [85] digest_0.6.37               BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [87] SparseArray_1.6.0           rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [89] htmlwidgets_1.6.4           farver_2.1.2               </span></span>
<span><span class="co">#&gt; [91] htmltools_0.5.8.1           pkgdown_2.1.1              </span></span>
<span><span class="co">#&gt; [93] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [95] microbenchmark_1.5.0        openssl_2.2.2              </span></span>
<span><span class="co">#&gt; [97] MASS_7.3-61</span></span></code></pre></div>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:mwoo5086@uni.sydney.edu.au" class="email">mwoo5086@uni.sydney.edu.au</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Max Woollard, Linh Nghiem, Shila Ghazanfar.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
